src/Models/Thread.php

```diff
@@ -153,6 +153,23 @@ public function participantsUserIds($userId = null)
         return $users->toArray();
     }
 
+    /**
+     * Thread relationship.
+     *
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @param object $object
+     *
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopePublicThread(Builder $query, $object)
```

Comment:
This now combines two things - a thread should be able to be public without an object same that a thread should be able to have a object without being public.

Let's take a car dealer:
* there is one public channel for all employees
* there could be 1:1 and group chats between employees
* there is a channel for every car-brand they sell - that's public (for all employees)
* the boss has a smart car and this channel has the boss-car as subject but is private cause no employee should read about the boss-car stats

This just as a basic example.

And cause of the fact that it just limits the query and doesn't limit to one it should be `Threads` with a plural *s*.
I learned it the hard way to prepend all scope methods with `by` to prevent conflicts with other methods - otherwise the `scopePublicThread()` is called the same like `publicThread()`.